reporters/osc_reporter.py

```diff
@@ -18,21 +18,31 @@ def flatten(self, root_path, data_dict):
                 pair = (param_path, param_value)
                 to_extend = [pair]
             else:
-                to_extend = self.flatten(param_path, param_value)
+                to_extend = self.flatten(param_value, param_path)
             pairs.extend(to_extend)
         
         return pairs
     
     def send(self, data_dict):
         # flatten dictionary into a list of pairs
-        send_pairs = self.flatten(OSC_BASE_PATH + BFI_ROOT, data_dict)
+        send_pairs = self.flatten(data_dict, OSC_BASE_PATH + BFI_ROOT)
 
         # send each pair
         for path, value in send_pairs:
             self.osc_client.send_message(path, value)
         
         return send_pairs
 
+class Debug_Reporter(OSC_Reporter):
+    def __init__(self, ip, send_port):
+        super().__init__(ip, send_port)
+    
+    def send(self, data_dict):
+        # flatten dictionary into a list of pairs
+        send_pairs = self.flatten(data_dict)
 
-
+        # send each pair
+        for path, value in send_pairs:
+            self.osc_client.send_message(path, value)
         
+        return send_pairs
```

Comment:
I would recommend moving this into its own file personally. It's generally considered good practice when writing object-oriented code to keep your files as single-object definitions